src/vivarium/interface/interactive.py

```diff
@@ -116,22 +117,24 @@ def replace_component(self, old_component, new_component):
         self.component_manager.add_components([new_component])
 
 
-def initialize_simulation(components, input_config=None, plugin_config=None):
+def initialize_simulation(components: List, input_config: Union[Mapping, ConfigTree]=None,
+                          plugin_config: Mapping=None) -> InteractiveContext:
     config = build_simulation_configuration()
     config.update(input_config)
     plugin_manager = PluginManager(plugin_config)
 
     return InteractiveContext(config, components, plugin_manager)
 
 
-def setup_simulation(components, input_config=None, plugin_config=None):
+def setup_simulation(components: List, input_config: Union[Mapping, ConfigTree]=None,
+                     plugin_config: Mapping=None) -> InteractiveContext:
     simulation = initialize_simulation(components, input_config, plugin_config)
     simulation.setup()
 
     return simulation
 
 
-def initialize_simulation_from_model_specification(model_specification_file):
+def initialize_simulation_from_model_specification(model_specification_file: str) -> InteractiveContext:
```

Comment:
We should probably only have one place that returns an actual Simulation context.  Drop line 144 and instead return `initialize_simulation(components, simulation_config, plugin_config)`

Reply:
I need the plugin manager to get the parser for components though.

Reply:
Gross.